extra/fuegogin/adaptor.go

```diff
@@ -109,14 +110,18 @@ type ginRouteRegisterer[T, B any] struct {
 	originalPath string
 }
 
+type GroupedRouter interface {
+	BasePath() string
+}
+
 func (a ginRouteRegisterer[T, B]) Register() fuego.Route[T, B] {
 	// We must register the gin handler first, so that the gin router can
 	// mutate the route path if it is a RouterGroup.
 	// This is because gin groups will prepend the group path to the route path itself.
 	a.ginRouter.Handle(a.route.Method, a.originalPath, a.ginHandler)
 
-	if _, ok := a.ginRouter.(*gin.RouterGroup); ok {
-		a.route.Path = a.ginRouter.(*gin.RouterGroup).BasePath() + a.route.Path
+	if grouped, ok := a.ginRouter.(GroupedRouter); ok && !slices.Contains([]string{"", "/"}, grouped.BasePath()) {
+		a.route.Path = ginToFuegoRoute(grouped.BasePath()) + a.route.Path
 	}
 
 	return a.route
```

Comment:
🤔

I feel doubtful about the hack you are using to check if grouped.BasePath is empty or /

slices.Contains use generics,it looks like calling superhero for nothing.

Also, the readability is not great.

Please consider this

```suggestion
	if grouped, ok := a.ginRouter.(GroupedRouter); ok {
		basePath := grouped.BasePath()
		switch basePath {
			case "", "/":
				// exclude these ones
			default:
				a.route.Path = ginToFuegoRoute(grouped.BasePath()) + a.route.Path
 	}
 
 	return a.route
```

Reply:
Done!